File: software/examples/c/example_stack_status.c
#include <stdio.h>

#include "ip_connection.h"
#include "brick_master.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XXYYZZ" // Change XXYYZZ to the UID of your Master Brick

int main(void) {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	Master master;
	master_create(&master, UID, &ipcon);

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		return 1;
	}
	// Don't use device before ipcon is connected

	// Get current stack voltage
	uint16_t stack_voltage;
	if(master_get_stack_voltage(&master, &stack_voltage) < 0) {
		fprintf(stderr, "Could not get stack voltage, probably timeout\n");
		return 1;
	}

	printf("Stack Voltage: %f V\n", stack_voltage/1000.0);

	// Get current stack current
	uint16_t stack_current;
	if(master_get_stack_current(&master, &stack_current) < 0) {
		fprintf(stderr, "Could not get stack current, probably timeout\n");
		return 1;
	}

	printf("Stack Current: %f A\n", stack_current/1000.0);

	printf("Press key to exit\n");
	getchar();
	master_destroy(&master);
	ipcon_destroy(&ipcon); // Calls ipcon_disconnect internally
	return 0;
}

File: software/examples/go/example_stack_status.go
package main

import (
	"fmt"
	"github.com/Tinkerforge/go-api-bindings/ipconnection"
	"github.com/Tinkerforge/go-api-bindings/master_brick"
)

const ADDR string = "localhost:4223"
const UID string = "XXYYZZ" // Change XXYYZZ to the UID of your Master Brick.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	master, _ := master_brick.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current stack voltage.
	stackVoltage, _ := master.GetStackVoltage()
	fmt.Printf("Stack Voltage: %f V\n", float64(stackVoltage)/1000.0)

	// Get current stack current.
	stackCurrent, _ := master.GetStackCurrent()
	fmt.Printf("Stack Current: %f A\n", float64(stackCurrent)/1000.0)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()
}
